### PR TITLE
Add document widget to event content type.

### PR DESCRIPTION
**Pull request by HelenaEksler**
#254

### CODE REVIEW

**Comment by HelenaEksler** [comment]
Update: Widget is working even if we have several document widgets in the form.
Still need design fixes.
widget_event

**Comment by HelenaEksler** [comment]
@amitaibu Can you review this please? And should I try to solve design problem in this PR or after merging? The order problem appears not only with this widget.

File: capacity4more/modules/c4m/widgets/c4m_document_widget/c4m_document_widget.module [hunk 1]
```diff
@@ -107,12 +106,17 @@ function c4m_document_widget_field_widget_form(&$form, &$form_state, $field, $in
...
     'basePath' => url('', $options),
     'csrfToken' => drupal_get_token(\RestfulInterface::TOKEN_VALUE),
     'data' => array(
       'purl' => url('', $options_context),
-      'relatedDocuments' => $default_values[1],
+      'relatedDocuments' => array(
+        $field_name => $default_values[1],
+      ),
+      // The form id in HTML is with dashes, not underscores.
+      'formId' => str_replace('_', '-', $form_state['build_info']['form_id']),
```
Comment (amitaibu) [inline capacity4more/modules/c4m/widgets/c4m_document_widget/c4m_document_widget.module]: drupal_clean_css_identifier ?

**Comment by HelenaEksler** [comment]
selection_160

**Comment by HelenaEksler** [comment]
selection_161

**Comment by HelenaEksler** [comment]
@amitaibu Fields are in the right places. PR is ready for the code review.

File: capacity4more/modules/c4m/widgets/c4m_document_widget/c4m_document_widget.module [hunk 2]
```diff
@@ -126,12 +134,58 @@ function c4m_document_widget_field_widget_form(&$form, &$form_state, $field, $in
...
+ *  The name of the field.
+ *
+ * @return String
+ *  Return fieldset name.
+ */
+function c4m_document_widget_get_field_parent_fieldset($entity_type, $field_name) {
+  $fieldset_name = NULL;
+  // Get parent fieldset name and add it as a property to the widget.
+  $field_groups = field_group_info_groups('node', $entity_type);
+  if (array_key_exists('form', $field_groups)) {
```
Comment (amitaibu) [inline capacity4more/modules/c4m/widgets/c4m_document_widget/c4m_document_widget.module]: I prefer ``!empty($field_groups['form'])``

File: capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
         <?php print t('Select a Document from the library') ?>
       </a>
     </div>
   </div>
+  <related-documents related-documents="data.relatedDocuments['<?php print $field_name; ?>']" form-id="formId" field-name="'<?php print $field_name; ?>'"></related-documents>
 
-  <related-documents related-documents="data.relatedDocuments"></related-documents>
-
-  <input type="text" id="related-documents" ng-hide="true" value="<?php print $default_value; ?>">
+  <input type="text" id="input-<?php print $field_name; ?>" ng-hide="true" value="<?php print $default_value; ?>">
```
Comment (amitaibu) [inline capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php]: if it's always hidden we don't need ng-hide, we can have a regular ``hidden`` class?

**Comment by HelenaEksler** [comment]
Merged manually into #334

**Pull request closed without merging**